### DISCUSSION TITLE
STM32F446VET6 grbl file

### DISCUSSION THREAD

**Discussion by ghost** [grblHAL/STM32F4xx#164] [Ideas]
Is it possible to have a grbl file to suit the STM32F446VET6 MCU?

This MCU has 100 pins, will run up to 180MHz and has up to 4 SPI's.

I am building a custom 6 Axis grbl board tha uses this MCU and will also utilise the HSI (instead of external crystal) to run at max frequency.

I am looking at (but am not sure if it is feasable) to run the SD Card on SPI 1 and to have a Wiznet Ethernet Module and ESP32 WiFi Module (which I am having discussions with you to build a driver) to run on SPI 4.

I have looked at the F446 Nucleo 64 (which I have previously used on another custom board) but I don't think this one will be appropriate. I can also see you have the Nucleo144 debug but I suspect that will not be appropriate either

I do not believe there is a commercial board that uses the 100 pin MCU.

Regards and have a great day
George

**Comment by terjeio** [comment]
> Is it possible to have a grbl file to suit the STM32F446VET6 MCU?

Yes, a clock tree setup may have to be added though - and perhas a linker file.

> I am looking at (but am not sure if it is feasable) to run the SD Card on SPI 1 and to have a Wiznet Ethernet Module and ESP32 WiFi Module (which I am having discussions with you to build a driver) to run on SPI 4.

Should be ok, but for the SD card you should really consider using SDIO if you are not constrained on pins.

If you want to run the WebUI then currently the plugin need more RAM than available, I may fix this later by swapping out the current json encoder with a new one. So a F7xx MCU with more than 128K RAM is perhaps a better choice?

**Reply by ghost** [reply]
I have started a new discussion within STM32F7XX repository for new MCU support

**Comment by ghost** [comment]
If I stick with SPI (for the SD Card) would that have any specific implications?

**Reply by terjeio** [reply]
Less transfer speed.

**Comment by ghost** [comment]
OK. Can the pin mappings for SDIO be changed  (use pins next to each other) and if yes, is this something that even I can do?

**Reply by terjeio** [reply]
Seems like not - there is only one SDMMC peripheral and it has fixed pin mappings. See the MCU datasheet.
And it is important to be careful when assigning input pins - once a pin number requiring interrupt capability is assigned the same pin number on another port cannot be used for an interrupt capable input. You want at least all the limit and control inputs to be assigned to separate pin numbers, and also the interrupt signals from WizNet/WiFi modules. If possible at least some aux inputs as well.
